test/rest_api/rest_util.py

```diff
@@ -0,0 +1,32 @@
+import time
+
+from contextlib import contextmanager
+
+# A utility function for creating a new temporary snapshot.
+# If no keyspaces are given, a snapshot is taken over all keyspaces and tables.
+# If no tables are given, a snapshot is taken over all tables in the keyspace.
+# If no tag is given, a unique tag will be computed using the current time, in milliseconds.
+# It can be used in a "with", as:
+#   with new_test_snapshot(cql, tag, keyspace, [table(s)]) as snapshot:
+# This is not a fixture - see those in conftest.py.
+@contextmanager
+def new_test_snapshot(rest_api, keyspaces=[], tables=[], tag=""):
```

Comment:
In my opinion (but I won't hold the series because of this), since this is a tool specifically for **snapshots**, it belongs in the only file that tests snapshots (test_storage_service.py) and we don't need a separate source file for it. I'm not a fan of used-once "utilities". But as I said, I won't insist on this.

Reply:
By the way, why does tag default to ""? Is this even legal?

Reply:
It defaults to `f"test_snapshot_{int(time.time() * 1000)}"` following the convention used elsewhere.
`new_test_snapshot` yields it as the result so it makes sense to leave it for the function if not provided by the user.